graphix/optimization.py

```diff
@@ -0,0 +1,77 @@
+"""Optimization procedures for patterns."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from graphix import Pattern, command
+from graphix.clifford import Clifford
+from graphix.command import CommandKind
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping
+    from collections.abc import Set as AbstractSet
+
+
+def _incorporate_pauli_results_in_domain(
+    results: Mapping[int, int], domain: AbstractSet[int]
+) -> tuple[bool, set[int]] | None:
+    result = None
+    odd = False
+    for node, outcome in results.items():
+        if node in domain:
+            if outcome:
+                odd = not odd
+            if result is None:
+                result = set(domain)
+            result.remove(node)
```

Comment:
Slightly more concise: (?) 

```suggestion
result = domain - results.keys() if domain & results.keys() else None
odd = 0
for i in domain & results.keys():
	odd ^= results[i]
```